src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -175,6 +181,301 @@ static inline ASR::asr_t* create_UnaryFunction(Allocator& al, const Location& lo
+{
+    /*
+     * Knuth-Morris-Pratt (KMP) string-matching
+     * This function takes two parameters:
+     *     the sub-string or pattern string and the target string,
+     * then returns the position of the first occurrence of the
+     * string in the pattern.
+     */
+    std::string fn_name = global_scope->get_unique_name("KMP_string_matching");
+    SymbolTable *fn_symtab = al.make_new<SymbolTable>(global_scope);
+
+    Vec<ASR::expr_t*> args;
+    ASR::ttype_t *char_type = TYPE(ASR::make_Character_t(al, loc, 1, -2,
+        nullptr, nullptr, 0));
+    {
+        args.reserve(al, 2);
+        create_var_expr(arg_1, "target_string", ASR::intentType::In,
+            ASR::abiType::Source, false, fn_symtab, char_type);
+        args.push_back(al, arg_1);
+        create_var_expr(arg_2, "pattern", ASR::intentType::In,
+            ASR::abiType::Source, false, fn_symtab, char_type);
+        args.push_back(al, arg_2);
+    }
+    ASR::ttype_t *int_type = TYPE(ASR::make_Integer_t(al, loc, 4, nullptr, 0));
+    create_variable(return_var, "result", ASRUtils::intent_return_var,
+        ASR::abiType::Source, false, fn_symtab, int_type);
+
+    Vec<ASR::stmt_t*> body;
+    body.reserve(al, 1);
+
+    SetChar dep;
+    dep.reserve(al, 1);
+    {
+        create_var_expr(pi_len, "pi_len", ASR::intentType::Local,
+            ASR::abiType::Source, false, fn_symtab, int_type);
+        create_var_expr(i, "i", ASR::intentType::Local, ASR::abiType::Source,
+            false, fn_symtab, int_type);
+        create_var_expr(j, "j", ASR::intentType::Local, ASR::abiType::Source,
+            false, fn_symtab, int_type);
+        create_var_expr(s_len, "s_len", ASR::intentType::Local,
+            ASR::abiType::Source, false, fn_symtab, int_type);
+        create_var_expr(pat_len, "pattern_len", ASR::intentType::Local,
+            ASR::abiType::Source, false, fn_symtab, int_type);
+        ASR::ttype_t *logical_type = TYPE(ASR::make_Logical_t(al, loc, 4,
+            nullptr, 0));
+        create_var_expr(flag, "flag", ASR::intentType::Local,
+            ASR::abiType::Source, false, fn_symtab, logical_type);
+        create_var_expr(lps, "lps", ASR::intentType::Local, ASR::abiType::Source,
+            false, fn_symtab, TYPE(ASR::make_List_t(al, loc, int_type)));
+        ASR::expr_t *result = EXPR(ASR::make_Var_t(al, loc, return_var));
+
+
+        body.push_back(al, STMT(ASR::make_Assignment_t(al, loc, s_len, EXPR(
+            ASR::make_StringLen_t(al, loc, args[0], int_type, nullptr)), nullptr)));
+        body.push_back(al, STMT(ASR::make_Assignment_t(al, loc, pat_len, EXPR(
+            ASR::make_StringLen_t(al, loc, args[1], int_type, nullptr)), nullptr)));
+        body.push_back(al, STMT(ASR::make_Assignment_t(al, loc, result,
+            EXPR(ASR::make_IntegerUnaryMinus_t(al, loc,
+            EXPR(ASR::make_IntegerConstant_t(al, loc, 1, int_type)), int_type,
+            EXPR(ASR::make_IntegerConstant_t(al, loc, -1, int_type)))), nullptr)));
```

Comment:
You can write an API to create constants like the way I am doing in, https://github.com/lfortran/lfortran/pull/1577